### DISCUSSION TITLE
Tauri Websocket Plugin V1

### DISCUSSION THREAD

**Discussion by mahadiResonest** [tauri-apps/tauri#10096] [Q&A] [answered]
I am intending to use tauri websocket plugin V1 for creating ws client and maintain it using rust process. I am using reactjs as frontend, and use websocket plugin api for communicating with the websocket. Now i saw there are four functions, 
1. Connect--> connects to ws server
2. addEventListener-->to listen event message from server
3. disconnect --> disconnect from client
4. send--> send message to server.

Now in my js i have a wrapper class that when the UI starts, i create a ws connection and keep it, all my components use this to send and listen message fron server. Now, my question is if for some reason the ws connection get disconnected (for network issue, or server stops) etc. How i know it? there is no onClose function or something. When the connection close, does my ws variable (where i kept the ws connection, is automatically updated to null or something?) I mean how i know that ws got disconnected?

**Comment by FabianLars** [comment, marked as answer]
It seems like the TypeScript type is incorrect but the websocket listener should also report errors. For more or less expected disconnects there's also a Close message.

tbh this didn't see much testing so we're not quite sure either...

**Comment by x93008** [comment]
This seems to be a basic feature, are there any plans to fix it

**Reply by FabianLars** [reply]
I can't justify spending time on the ws plugin (especially v1) at the moment so no plans on our side atm and I don't expect anyone to contribute either sadly.

Also what exactly is the issue you're having? Is  my other comment incorrect?
